src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -240,6 +241,13 @@ def _append_uid(self, user):
                 if user['tenant'] else user['user_id']
         return user
 
+    @staticmethod
+    def _keys_allowed():
+        permissions = AuthManager.get_user(JwtManager.get_username()).permissions_dict()
+        edit_permissions = [Permission.CREATE, Permission.UPDATE, Permission.DELETE]
+        return Scope.RGW in permissions and Permission.READ in permissions[Scope.RGW] \
+            and len(set(edit_permissions).intersection(set(permissions[Scope.RGW]))) > 0
```

Comment:
I wouldn't complicate things here: R(ead) is always provided in our roles. If no Read, it's logical to assume no CRD access can be effectively provided (kind of redundant, but that's it).
```py
try:
  return Permission.READ in permissions[Scope.RGW]
except KeyError...:
```

Reply:
This is for predefined roles, but you can create a custom role with no read but other permission
(it would be weird, but it's doable) so I prefer to keep the check here.

Reply:
Yup, but if we think that kind of customization is going to be used, then we should be fully consistent with the expectations. For example, if some role is given `DELETE` permission, I wouldn't assume it has access to existing keys (you can create an RGW user without unveiling its keys). And I'd say the same applies to 'CREATE' or 'UPDATE' permissions (new keys provided shouldn't require knowing the previous ones). So strictly speaking, only 'READ' permission should provide access to reading keys.

Reply:
This is done accordingly to what was discussed on the issue itself:
https://tracker.ceph.com/issues/42475#note-4
A read-only user then can see API keys provided it has some kind of edit privileges.
Otherwise, there is no real bug to fix.

Reply:
Please ignore my previous comment. I forgot the context of the issue: we are trying to hide sensitive information to 'read-only' users, as this information could allow them (via CLI) to modify or delete items (hence, overrunning their intended permissions).